evalml/data_checks/uniqueness_data_check.py

```diff
@@ -0,0 +1,105 @@
+            ... })
+            >>> uniqueness_check = UniquenessDataCheck(problem_type="regression", threshold=0.8)
+            >>> assert uniqueness_check.validate(df) == {"errors": [],\
+                                                         "warnings": [{"message": "Input columns (regression_not_unique_enough) for regression problem type are not unique enough.",\
+                                                                 "data_check_name": "UniquenessDataCheck",\
+                                                                 "level": "warning",\
+                                                                 "code": "NOT_UNIQUE_ENOUGH",\
+                                                                 "details": {"column": "regression_not_unique_enough", 'uniqueness_score': 0.0}}]}
+        """
+        messages = {
+            "warnings": [],
+            "errors": []
+        }
+
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        res = X.apply(UniquenessDataCheck.uniqueness_score)
+
+        if is_regression(self.problem_type):
+            not_unique_enough_cols = list(res.index[res < self.threshold])
+            messages["warnings"].extend([DataCheckWarning(message=warning_not_unique_enough.format(col_name,
+                                                                                                   self.problem_type),
+                                                          data_check_name=self.name,
+                                                          message_code=DataCheckMessageCode.NOT_UNIQUE_ENOUGH,
+                                                          details={"column": col_name, "uniqueness_score": res.loc[col_name]}).to_dict()
+                                         for col_name in not_unique_enough_cols])
+        elif is_multiclass(self.problem_type):
+            too_unique_cols = list(res.index[res > self.threshold])
+            messages["warnings"].extend([DataCheckWarning(message=warning_too_unique.format(col_name,
+                                                                                            self.problem_type),
+                                                          data_check_name=self.name,
+                                                          message_code=DataCheckMessageCode.TOO_UNIQUE,
+                                                          details={"column": col_name, "uniqueness_score": res.loc[col_name]}).to_dict()
+                                         for col_name in too_unique_cols])
+        return messages
+
+    @staticmethod
+    def uniqueness_score(col):
+        """This function calculates a uniqueness score for the provided field.
+
+        Arguments:
+            col (pd.Series): Feature values.
+        Returns:
+            (float): Uniqueness score.
+        """
+        norm_counts = col.value_counts() / col.value_counts().sum()
```

Comment:
nit-pick, but is `col.value_counts().sum()` the same as `len(col)`? Is there a reason for doing it this way?

Reply:
This is a good point.  I added another test (as I was curious myself).  And I think invoking value_counts is nicer because it ignores NaNs.  In the case that you have NaN in your column, len(cols) != col.value_counts().sum() as the RHS will be lower.